### ISSUE TITLE
Can we use other ensemble decision tree?

### ISSUE THREAD

**Issue by jermaine1ronquillo** [parrt/dtreeviz#15]
Can we use Catboost, XGBoost, and Lightgbm?

**Comment by parrt** [comment]
Not currently. You would just have to get ShadowDecisionTree wrappers for those trees. Note that boosting trees are usually very shallow; not sure it'll help but try it out!

**Comment by jermaine1ronquillo** [comment]
Thank you for the response

**Comment by hcho3** [comment]
This is a really cool project. It would be interesting to see XGBoost integrated into dtreeviz.

>  You would just have to get ShadowDecisionTree wrappers for those trees. 

So I’d just need to define a new class that looks like ShadowDecTree? I have some experience parsing XGBoost models; let me see if I can make a working pull request.

**Comment by parrt** [comment]
@hcho3 yep, would be nice. I would consider a PR. :)

**Comment by martin-loetzsch** [comment]
@hcho3 Have you been able to write that wrapper class for XGBoost? And if so, can you share it?

**Comment by H4dr1en** [comment]
Can we re-open? A support for XGBoost would be very interested!
I am pretty sure it is  feasible using the [trees_to_dataframe](https://github.com/dmlc/xgboost/blob/278562db133005f93edbffc74afc4b02b49bccca/python-package/xgboost/core.py#L1583) function to get a dataframe description of the trees:

![image](https://user-images.githubusercontent.com/16240134/58461602-579f9400-8130-11e9-8b97-b2b5f1b0a08f.png)

**Comment by parrt** [comment]
Ah. ok. maybe somebody could submit a PR.

**Comment by Jacky2Wong** [comment]
Is there a way to determine which tree is the best to show by default? I was thinking ideally the one with the minimum loss value but I am not sure if XGBoost has a method that tells us which tree this is.

**Comment by tlapusan** [comment]
hi @hcho3 ,  @martin-loetzsch , @H4dr1en 
we've just merged xgboost implementation into master. Before making a new release, would be helpful to have a feedback from your sides.

You can track more comments from https://github.com/parrt/dtreeviz/issues/83 and https://github.com/parrt/dtreeviz/pull/91

For more details about visualizations, you can take a look at https://github.com/parrt/dtreeviz/blob/master/notebooks/dtreeviz_xgboost_visualisations.ipynb

Thanks, Tudor.

**Comment by hcho3** [comment]
I filed a pull request in the XGBoost repository to mention dtreeviz: https://github.com/dmlc/xgboost/pull/6013. Thank you for the great work!

**Comment by tlapusan** [comment]
Thanks @hcho3 for your support. Is there anything we can help with the demo, or is something internal to xgboost team?

**Comment by hcho3** [comment]
@tlapusan Mainly we want to put up a self-contained demo script that can be automatically run by our CI system. That way, any future changes of XGBoost won't break dtreeviz accidentally.

**Comment by parrt** [comment]
closing this as we have xgboost and @tlapusan  is working on lightgbm
